packages/commonwealth/client/scripts/views/modals/unified_user_flow_modal.tsx

```diff
@@ -0,0 +1,268 @@
+/* @jsx m */
+
+import ClassComponent from 'class_component';
+import { notifyError } from 'controllers/app/notifications';
+import $ from 'jquery';
+import m from 'mithril';
+
+import 'modals/unified_user_flow_modal.scss';
+
+import app from 'state';
+import { CWButton } from '../components/component_kit/cw_button';
+import { CWIcon } from '../components/component_kit/cw_icons/cw_icon';
+import { CWText } from '../components/component_kit/cw_text';
+import { CWTextInput } from '../components/component_kit/cw_text_input';
+
+class EntryPage extends ClassComponent<{
+  onCancel: () => void;
+  onSave: () => void;
+}> {
+  private displayNameValue = '';
+  private validatedName = false;
+  view(vnode) {
+    const { onSave, onCancel } = vnode.attrs;
+
+    return (
+      <div class="EntryModal">
+        <div class="Header">
+          <CWText type="caption" fontWeight="bold">
+            INTRODUCING
+          </CWText>
+          <CWText type="h3" fontWeight="bold">
+            Unified User Profiles
+          </CWText>
+        </div>
+        <div class="Description">
+          <CWText type="b1">
+            With this update you'll have a single display name across all your
+            communities, making it easier to connect and engage with others.
+          </CWText>
+          <CWText type="b1" className="bold-section">
+            This change will go into effect
+            <span class="bold"> March 8th, 2023. </span>
+            To prepare, provide a display name below.
+          </CWText>
+        </div>
+        <div class="Inputs">
+          <CWText type="caption">Display Name</CWText>
+          <CWTextInput
+            placeholder="ex: Common Cow"
+            value={this.displayNameValue}
+            oninput={(e) => {
+              this.displayNameValue = e.target.value;
+            }}
+            inputValidationFn={(val) => {
+              // eslint-disable-next-line no-useless-escape
+              const regex = /^([a-zA-Z0-9\s\_\-]+)$/;
```

Comment:
Replace `\s` with ` ` -- we don't want to allow tabs and carriage returns, just plain old spaces.

Reply:
Not addressed.

Reply:
`^([a-zA-Z0-9 \_\-]+)$` (all i did was replace `\s` with ` `)